### DISCUSSION TITLE
LDAP - Bind error 49  (Invalid credentials),

### DISCUSSION THREAD

**Discussion by Lulan1999** [ltb-project/self-service-password#836] [Q&A] [answered]
i need help that works perfectly i can connect to ldap
 ldapsearch -D "cn=lonalu ,cn=Users,dc=zuzu,dc=local" -x -w "password" -h 10.0.0.139 -b "cn=Users,dc=zuzu,dc=local"

but configuration what i have is showing that problem  LDAP - Bind error 49  (Invalid credentials),  when try to do anything on site as change password or send e-mail, anyone could explain me where is the problem? what do i do wrong? i am using docker version, configuration for sure is working ,because i tested on 2 diffrent ad one on linux and one on windows when i changed ip i had two difrent messages but now i stuck on that problem. please give me some advice
$ldap_url = "ldap://10.0.0.139";
$ldap_binddn = "cn=lonalu ,ou=Users,dc=zuzu,dc=local";
$ldap_bindpw = "password";
$keyphrase = "mysecret"; 
$ldap_starttls = false;
$ldap_base = "dc=zuzu,dc=local";
$debug = true;

whem i leave binddn and bindpw empty error is 
 LDAP - Search error 1 (Operations error), referer

**Comment by Lulan1999** [comment]
anyone can help me?

**Comment by davidcoutadeur** [comment]
It's quite simple, you only have 3 options:
- your dn is incorrect: cn=lonalu ,ou=Users,dc=zuzu,dc=local
- your password is incorrect: password
- you think your dn or password are the values defined above, but actually they are not.

**Comment by Lulan1999** [comment]
i found solution, documentation is wrong i have to change 

cn=lonalu ,ou=Users,dc=zuzu,dc=local to
cn=lonalu, cn=Users,dc=zuzu,dc=local and i can connect but now issue is diffrent 

 LDAP - User lonalu not found 

$ldap_base = "dc=zuzu,dc=local"; is set to this. why i cannot find any user ? any user i am trying to find i cannot

**Comment by Lulan1999** [comment]
because i changed cn=lonalu ,ou=Users,dc=zuzu,dc=local to cn=lonalu, cn=Users,dc=zuzu,dc=local i am able to connect but when i am trying to send e-mail for change password i m recieving  information that LDAP - User lonalu not found anyone can tell me what is wrong? my configuration e-mail is not set yet i am aware of that. 

php // My SSP configuration
$ldap_url = "ldap://10.0.0.139";
$ldap_binddn = "cn=lonalu,cn=Users,dc=zuzu,dc=lan";
$ldap_bindpw = "password";
$keyphrase = "mysecret";
$ldap_starttls = false;
$ldap_base = "dc=zuzu,dc=lan";
$debug = true;
$ldap_filter = "(&(objectClass=person)(uid={login}))";

when i connect by ldapsearch -D "cn=lonalu ,cn=Users,dc=zuzu,dc=local" -x -w "password" -h 10.0.0.139 -b "cn=Users,dc=zuzu,dc=local" 
response i recieved 

 lonalu , Users, zuzu.local
dn: CN=lonalu,CN=Users,DC=zuzu,DC=local
objectClass: top
objectClass: person
objectClass: organizationalPerson
objectClass: user
cn: lonalu 
sn: lonalu 
givenName: lonalu 
distinguishedName: CN=lonalu ,CN=Users,DC=zuzu,DC=local
instanceType: 4
whenCreated: 20240219104518.0Z
whenChanged: 20240219104605.0Z
displayName: lonalu 
uSNCreated: 19173
memberOf:: Q049VHfDs3JjeS13xYJhxZtjaWNpZWxlIHphc2FkIGdydXB5LENOPVVzZXJzLERDPXp
 zcG0sREM9bGFu
memberOf: CN=Administrators domain,CN=Users,DC=zuzu,DC=local
memberOf:: Q049QWRtaW5pc3RyYXRvcnp5IHByemVkc2nEmWJpb3JzdHdhLENOPVVzZXJzLERDPXp
 zcG0sREM9bGFu
memberOf: CN=Administrators schematu,CN=Users,DC=zuzu,DC=local
memberOf: CN=Administrators,CN=Builtin,DC=zuzu,DC=local
uSNChanged: 19195
name: lonalu 
objectGUID:: gSILnAKgLUa1DF26oNtLAg==
userAccountControl: 66048
badPwdCount: 0
codePage: 0
countryCode: 0
badPasswordTime: 0
lastLogoff: 0
lastLogon: 0
pwdLastSet: 133528131180504330
primaryGroupID: 513
objectSid:: AQUAAAAAAAUVAAAALnxH/gL9BBzfO3JTUgQAAA==
accountExpires: 9223372036854775807
logonCount: 0
sAMAccountName: lonalu 
sAMAccountType: 805306368
userPrincipalName: lonalu@zuzu.local
objectCategory: CN=Person,CN=Schema,CN=Configuration,DC=zuzu,DC=local
dSCorePropagationData: 16010101000000.0Z
lastLogonTimestamp: 133528131650987761

**Reply by coudot** [reply]
There is no uid in your entry.

Please read documentation: https://self-service-password.readthedocs.io/en/latest/config_ldap.html#active-directory

**Reply by Lulan1999** [reply]
could you send me an example in my case ?

**Reply by davidcoutadeur** [reply, marked as answer]
Hello @Lulan1999 

On one hand, if you are stuck on a specific problem, especially if you think it can help the whole community or improve the software, do not hesitate to send your questions on the mailing list.

On the other hand, if you just need to configure your software, I only see two options:
- reading the documentation (see the URL pointed by @coudot )
- asking for a professional service. You have a list of companies that might be able to help you at this page: https://ltb-project.org/professionalservices.html

**Reply by Lulan1999** [reply]
thank you for your advice, i am slowly going forward with that problem all the time pop up new one but those small helps already let me think what should i do ! you are the best guys
